src/extension.ts

```diff
@@ -272,20 +294,68 @@ class RdbgAdapterDescriptorFactory implements DebugAdapterDescriptorFactory {
 
 	make_shell_command(cmd: string) {
 		const shell = process.env.SHELL;
-		switch (true) {
-			case shell && (shell.endsWith("bash") || shell.endsWith("fish")):
-				return shell + " -l -c '" + cmd + "'";
-			case shell && shell.endsWith("zsh"):
-				// As the recommended way, initialization commands for rbenv are written in ".zshrc".
-				// However, it's not loaded on the non-interactive shell.
-				// Thus, we need to run this command as the interactive shell.
-				// FYI: https://zsh.sourceforge.io/Guide/zshguide02.html
-				return shell + " -l -c -i '" + cmd + "'";
-			default:
-				return cmd;
+
+		if (this.support_login(shell)) {
+			return shell + " -lic '" + cmd + "'";
+		} else {
+			return cmd;
```

Comment:
Can you cut out this part to another PR? I can merge this part soon.

Reply:
Why not merge this PR instead?

Reply:
Sorry, I can't merge this PR since @ko1-san is reviewing this PR, and he decides it. And the reason why I ask you to cut this part to another PR is that it is significant to merge this part regardless of this feature.

Reply:
But, it's just a suggestion, I'll leave it up to you.

Reply:
I understand your point of view, but my sincere opinion is that we should focus on getting this PR merged.

Using Ruby version managers is common practice on the community and making sure the environment is activated properly is essential to get all of the benefits of RDBG in the editor.